securedrop/server.py

```diff
@@ -235,6 +265,52 @@ async def add_contact(self, addc, stream):
         msg = self.users.add_contact(self.sock_to_email[stream], addc.name, addc.email)
         await self.write_status(stream, msg)
 
+    # 1. `X -> Y/F -> S`: X wants to send F to Y
+    async def process_file_transfer_request(self, ftrp, stream):
+        sender_email = self.sock_to_email[stream]
+        recipient_email = ftrp.recipient_email
+        msg = ""
+        if recipient_email not in self.email_to_sock:
+            msg = "User [{}] is not online".format(recipient_email)
+        elif not self.users.contacts_contains(recipient_email, sender_email):
+            msg = "User [{}] has not added you as a contact".format(recipient_email)
```

Comment:
This is a bit confusing. I think adding a brief comment here would make this a bit easier to understand.

Reply:
What is confusing about it?

Reply:
It was actually this line. I was confused by it at first and thought it could use a comment but on a second look it makes more sense:
```
self.sock_to_address[stream][0] != self.sock_to_address[self.email_to_sock[recipient_email]][0]:
```